container/lxd/initialisation_linux.go

```diff
@@ -304,47 +349,39 @@ func ensureDependencies(series string) error {
 //
 // TODO(frobware): this only caters for IPv4 setups.
 func findNextAvailableIPv4Subnet() (string, error) {
-	_, ip10network, err := net.ParseCIDR("10.0.0.0/16")
+	knownIPs, knownCIDRs, err := getKnownV4IPsAndCIDRs(interfaceAddrs)
 	if err != nil {
 		return "", errors.Trace(err)
 	}
 
-	addrs, err := interfaceAddrs()
-	if err != nil {
-		return "", errors.Annotatef(err, "cannot get network interface addresses")
-	}
-
-	max := 0
-	usedSubnets := make(map[int]bool)
-
-	for _, address := range addrs {
-		addr, network, err := net.ParseCIDR(address.String())
+	randomized3rdSegment := randomizedOctetRange()
+	for _, i := range randomized3rdSegment {
+		// lxd randomizes the 2nd and 3rd segments, we should be fine with the
+		// 3rd only
+		ip, ip10network, err := net.ParseCIDR(fmt.Sprintf("10.0.%d.0/24", i))
 		if err != nil {
-			logger.Debugf("cannot parse address %q: %v (ignoring)", address.String(), err)
-			continue
+			return "", errors.Trace(err)
 		}
-		if !ip10network.Contains(addr) {
-			logger.Debugf("find available subnet, skipping %q", network.String())
-			continue
+
+		collides := false
+		for _, kIP := range knownIPs {
+			if ip10network.Contains(kIP) {
+				collides = true
+				break
+			}
 		}
-		subnet := int(network.IP[2])
-		usedSubnets[subnet] = true
-		if subnet > max {
-			max = subnet
+		if !collides {
+			for _, kNet := range knownCIDRs {
+				if kNet.Contains(ip) || ip10network.Contains(kNet.IP) {
+					collides = true
```

Comment:
https://github.com/juju/juju/pull/7054/files#diff-04db2fba070a0bd3ebfc387dfc486967R315 is the test I updated to test that we handle colliding values.